spring-aop/src/main/java/org/springframework/aop/TargetClassAware.java

```diff
@@ -29,6 +29,7 @@
  * @since 2.0.3
  * @see org.springframework.aop.support.AopUtils#getTargetClass(Object)
  */
+@FunctionalInterface
```

Comment:
I don't think this is required really. Just because an interface has a single method doesn't mean it's a functional interface. These callbacks are not meant to be used with lambdas so I don't think this should be done.